src/uint/rand.rs

```diff
@@ -30,18 +30,29 @@ impl<const LIMBS: usize> RandomMod for Uint<LIMBS> {
     /// issue so long as the underlying random number generator is truly a
     /// CSRNG, where previous outputs are unrelated to subsequent
     /// outputs and do not reveal information about the RNG's internal state.
-    fn random_mod(mut rng: &mut impl CryptoRngCore, modulus: &NonZero<Self>) -> Self {
+    fn random_mod(rng: &mut impl CryptoRngCore, modulus: &NonZero<Self>) -> Self {
         let mut n = Self::ZERO;
 
         let n_bits = modulus.as_ref().bits_vartime();
+        let n_bytes = (n_bits + 7) / 8;
         let n_limbs = (n_bits + Limb::BITS - 1) / Limb::BITS;
-        let mask = Limb::MAX >> (Limb::BITS * n_limbs - n_bits);
+        let hi_bytes = n_bytes - (n_limbs - 1) * Limb::BYTES;
+
+        let mut bytes = Limb::ZERO.to_le_bytes();
 
         loop {
-            for i in 0..n_limbs {
-                n.limbs[i] = Limb::random(&mut rng);
+            for i in 0..n_limbs - 1 {
+                rng.fill_bytes(bytes.as_mut());
+                // Need to deserialize from little-endian to make sure that two 32-bit limbs
+                // deserialized sequentially are equal to one 64-bit limb produced from the same
+                // byte stream.
+                n.limbs[i] = Limb::from_le_bytes(bytes);
             }
-            n.limbs[n_limbs - 1] = n.limbs[n_limbs - 1] & mask;
```

Comment:
Have you evaluated the performance impact of removing the masking?

Reply:
I don't think the impact is noticeable compared to replacing `next_u64()/next_u32()` (which were called inside `Limb::random()`) with `fill_bytes()`. I need to be sure I'm drawing the same number of bytes from the RNG on every platform, so masking does not really fit into the algorithm.

Reply:
Now, an RNG implementation may actually skip bytes differently on every platform between successive calls to `fill_bytes()` - this is allowed as long as the distribution remains random. So this implementation is only consistent as long as the RNG is, and an RNG is not guaranteed to be, so perhaps we shouldn't provide that guarantee at all.

In my application I am now using an XOF instead, anyway; this PR is only to protect other uses from the same issue I encountered, but perhaps it's better to make an explicit disclaimer in the docs instead.